Source/WebCore/platform/graphics/Damage.h

```diff
@@ -46,11 +46,26 @@ class Damage {
 
     static constexpr int s_tileSize { 256 };
 
-    Damage()
-        : m_tileSize(s_tileSize, s_tileSize)
+    enum class Mode : uint8_t {
+        Rectangles, // Tracks dirty regions as rectangles, only unifying when maximum is reached.
+        BoundingBox, // Dirty region is always the minimum bounding box of all added rectangles.
+        Full, // All area is always dirty.
+    };
+
+    explicit Damage(const IntSize& size, Mode mode = Mode::Rectangles)
+        : m_mode(mode)
+        , m_size(size)
+        , m_tileSize(s_tileSize, s_tileSize)
     {
-        // FIXME: Add a constructor to pass the size.
-        resize(IntSize { 2048, 1024 });
+        initialize();
+    }
+
+    explicit Damage(const FloatSize& size, Mode mode = Mode::Rectangles)
+        : m_mode(mode)
+        , m_size(ceiledIntSize(LayoutSize(size)))
+        , m_tileSize(s_tileSize, s_tileSize)
+    {
+        initialize();
     }
```

Comment:
```suggestion
    explicit Damage(const FloatSize& size, Mode mode = Mode::Rectangles)
        : Damage(ceiledIntSize(LayoutSize(size)), mode)
    {
    }
```

we can do delegated constructors instead.